components/style/values/specified/box.rs

```diff
@@ -7,15 +7,194 @@
 use Atom;
 use cssparser::Parser;
 use parser::{Parse, ParserContext};
+#[cfg(feature = "servo")]
+use properties::{longhands, PropertyDeclaration};
 use std::fmt;
 use style_traits::{ParseError, ToCss};
 use values::CustomIdent;
 use values::KeyframesName;
+#[cfg(feature = "servo")]
+use values::computed::Context;
 use values::generics::box_::AnimationIterationCount as GenericAnimationIterationCount;
 use values::generics::box_::VerticalAlign as GenericVerticalAlign;
 use values::specified::{AllowQuirks, Number};
 use values::specified::length::LengthOrPercentage;
 
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug, Eq, Hash, MallocSizeOf, Parse, PartialEq, ToComputedValue, ToCss)]
+#[cfg_attr(feature = "servo", derive(Deserialize, Serialize))]
+/// Defines an element’s display type, which consists of
+/// the two basic qualities of how an element generates boxes
+/// <https://drafts.csswg.org/css-display/#propdef-display>
+pub enum Display {
+    Inline, Block, InlineBlock,
+    Table, InlineTable, TableRowGroup, TableHeaderGroup,
+    TableFooterGroup, TableRow, TableColumnGroup,
+    TableColumn, TableCell, TableCaption, ListItem, None,
+    #[parse(aliases = "-webkit-flex")]
+    Flex,
+    #[parse(aliases = "-webkit-inline-flex")]
+    InlineFlex,
+    #[cfg(feature = "gecko")]
+    Grid, InlineGrid, Ruby, RubyBase, RubyBaseContainer, RubyText, RubyTextContainer,
+    #[cfg(feature = "gecko")]
+    Contents, FlowRoot, WebkitBox, WebkitInlineBox,
+    #[cfg(feature = "gecko")]
+    MozBox, MozInlineBox, MozGrid, MozInlineGrid, MozGridGroup,
+    #[cfg(feature = "gecko")]
+    MozGridLine, MozStack, MozInlineStack, MozDeck, MozPopup, MozGroupbox,
+}
+
+impl Display {
+    /// The initial display value.
+    #[inline]
+    pub fn inline() -> Self {
+        Display::Inline
+    }
+
+    /// Returns whether this "display" value is the display of a flex or
+    /// grid container.
+    ///
+    /// This is used to implement various style fixups.
+    pub fn is_item_container(&self) -> bool {
+        match *self {
+            Display::Flex | Display::InlineFlex => true,
+            #[cfg(feature = "gecko")]
+            Display::Grid | Display::InlineGrid => true,
+            _ => false,
+        }
+    }
+
+    /// Returns whether an element with this display type is a line
+    /// participant, which means it may lay its children on the same
+    /// line as itself.
+    pub fn is_line_participant(&self) -> bool {
+        match *self {
+            Display::Inline => true,
+            #[cfg(feature = "gecko")]
+            Display::Contents | Display::Ruby
+                | Display::RubyBaseContainer => true,
```

Comment:
nit: indentation here looks somewhat weird, what about:

```rust
    pub fn is_line_participant(&self) -> bool {
        match *self {
            Display::Inline => true,
            #[cfg(feature = "gecko")]
            Display::Contents |
            Display::Ruby |
            Display::RubyBaseContainer => true,
```

Not particularly sold on it, so if you think it's cleaner as-is that's fine for me too.

Reply:
I was thinking about

```rust
match *self {
  Display::Inline => true,
  #[cfg(feature = "gecko")]
  Display::Contents
    | Display::Ruby
    | Display::RubyBaseContainer  => true,
```

but waiting for your comments :P

I'll follow the indentation you provided above, thanks!.

Reply:
That works for me :)